visualization/open3d_vis_utils.py

```diff
@@ -0,0 +1,149 @@
+"""
+Utilities for rendering camera frustums and 3d point clouds using Open3d.
+
+Author: John Lambert
+"""
+
+import argparse
+from typing import List
+
+import numpy as np
+import open3d
+from colour import Color
+from gtsam import Cal3Bundler, Pose3
+
+from gtsfm.common.view_frustum import ViewFrustum
+
+
+def create_colored_point_cloud_open3d(point_cloud: np.ndarray, rgb: np.ndarray) -> open3d.geometry.PointCloud:
+    """Render a point cloud as individual colored points, using Open3d.
+
+    Args:
+        point_cloud: array of shape (N,3) representing 3d points.
+        rgb: uint8 array of shape (N,3) representing colors in RGB order, in the range [0,255]
+
+    Returns:
+        pcd: Open3d geometry object representing a colored 3d point cloud.
+    """
+    colors = rgb.astype(np.float64) / 255
+
+    pcd = open3d.geometry.PointCloud()
+    pcd.points = open3d.utility.Vector3dVector(point_cloud)
+    pcd.colors = open3d.utility.Vector3dVector(colors)
+
+    return pcd
+
+
+def create_colored_spheres_open3d(
+    sphere_radius: float, point_cloud: np.ndarray, rgb: np.ndarray
+) -> List[open3d.geometry.TriangleMesh]:
+    """Create a colored sphere mesh for every point inside the point cloud, using Open3d.
+
+    Note: this is quite computationally expensive.
+
+    Args:
+        sphere_radius: radius of each rendered sphere.
+        point_cloud: array of shape (N,3) representing 3d points.
+        rgb: uint8 array of shape (N,3) representing colors in RGB order, in the range [0,255]
+
```

Comment:
Move point_cloud and rgb to first two args?

Reply:
Because you could potentially have a default radius.

Reply:
good point, updated

Reply:
documentation has not been updated.

Reply:
just fixed it, thanks